core/src/core/pmmr.rs

```diff
@@ -92,82 +92,49 @@ where
 	fn dump_stats(&self);
 }
 
-/// Maixmum peaks for a Merkle proof
+/// Maximum peaks for a Merkle proof
 pub const MAX_PEAKS: u64 = 100;
 
-/// Maixmum path for a Merkle proof
+/// Maximum path for a Merkle proof
 pub const MAX_PATH: u64 = 100;
 
-/// A Merkle proof.
-/// Proves inclusion of an output (node) in the output MMR.
-/// We can use this to prove an output was unspent at the time of a given block
-/// as the root will match the output_root of the block header.
-/// The path and left_right can be used to reconstruct the peak hash for a given tree
-/// in the MMR.
-/// The root is the result of hashing all the peaks together.
-#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd, Serialize, Deserialize)]
+/// Merkle proof errors.
+#[derive(Clone, Debug, PartialEq)]
+pub enum MerkleProofError {
+	/// Merkle proof root hash does not match when attempting to verify.
+	RootMismatch,
+}
+
+/// A Merkle proof that proves a particular element exists in the MMR.
+#[derive(Debug, Eq, PartialEq, Clone, PartialOrd, Ord)]
 pub struct MerkleProof {
```

Comment:
Should we take this as an opportunity to move Merkle proofs in their own file? I think it'd be better to not clutter `pmmr.rs` too much.